lifecycle/lifecycle/server/metrics.py

```diff
@@ -24,8 +28,26 @@ def collect(self):
 
 def is_database_connected() -> bool:
     try:
+        DJANGO_DB_TYPE = os.environ.get('DJANGO_DB_TYPE', 'sqlite')
+        if DJANGO_DB_TYPE == 'postgres':
+            host = settings.DATABASES['default']['HOST']
+            port = settings.DATABASES['default']['PORT']
+            if not is_port_open(host, int(port)):
+                return False
+
         close_old_connections()
         connection.cursor().execute("select 1")
         return True
     except OperationalError:
         return False
+
+
+def is_port_open(ip: str, port: int) -> bool:
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.settimeout(3)
+    try:
+        s.connect((ip, port))
+        s.shutdown(2)
```

Comment:
`shutdown(2)` is a little opaque if you aren't aware of what that means already. I prefer the following
```suggestion
        s.shutdown(socket.SHUT_RDWR)
```

Reply:
Good point, thanks.